api.py

```diff
@@ -44,83 +44,145 @@ async def status():
+            json_to_validate = decode_and_parse_json(data)
+    else:
+        logger.error("No JSON data provided for validation", status=400)
+        return Response(status_code=400, content="No JSON data provided for validation")
 
     response = {}
     try:
+        logger.debug(
+            "Sending JSON data to AJV Schema Validator service...",
+            url=AJV_VALIDATOR_URL,
+        )
+        # Posts JSON data to AJV Validator service and returns a response containing any errors
         ajv_response = requests.post(
             AJV_VALIDATOR_URL, json=json_to_validate, timeout=10
         )
+        # Returns errors in the response if AJV Validator service returned any errors
         if ajv_response_dict := ajv_response.json():
             response["errors"] = ajv_response_dict["errors"]
-            logger.info("Schema validator returned errors", status=400)
+            logger.warning(
+                "AJV Schema Validator service returned errors",
+                status=400,
+                errors=response["errors"],
+            )
             return response, 400
 
     except RequestException:
-        logger.info("AJV Schema validator service unavailable")
-        return json.dumps(obj={}, error="AJV Schema validator service unavailable")
+        logger.error("AJV Schema Validator service unavailable")
+        return json.dumps(obj={}, error="AJV Schema Validator service unavailable")
+
+    logger.info("AJV Schema Validator service returned no errors", status=200)
 
     validator = QuestionnaireValidator(json_to_validate)
+    logger.debug(
+        "Attempting to validate questionnaire schema contents with Questionnaire Validator...",
+        questionnaire_title=json_to_validate.get("title"),
+    )
+    # Validates questionnaire schema contents using the QuestionnaireValidator
     validator.validate()
 
+    # Adds errors from validation to the response if there are any
     if validator.errors:
         response["errors"] = validator.errors
-        logger.info("Questionnaire validator returned errors", status=400)
+        logger.warning(
+            "Questionnaire Validator returned errors",
+            status=400,
+            errors=response["errors"],
+        )
         response = Response(content=json.dumps(response), status_code=400)
 
         return response
 
-    logger.info("Schema validation passed", status=200)
+    logger.info("Schema validation successfully completed with no errors", status=200)
 
     response = Response(content=json.dumps(response), status_code=200)
 
     return response
 
 
 def is_domain_allowed(parsed_url, domain):
+    logger.debug("Checking if domain is allowed...", domain=domain)
     base_url = f"{parsed_url.scheme}://{parsed_url.netloc}/"
     repo_owner = (
         parsed_url.path.split("/")[1] if len(parsed_url.path.split("/")) > 1 else ""
     )
+    logger.debug("Parsed URL components", base_url=base_url, repo_owner=repo_owner)
     return (
```

Comment:
We could be a little more explicit here and log what exactly happened so:
<img width="859" height="501" alt="Untitled 7" src="https://github.com/user-attachments/assets/19a3b580-2e67-4bc0-9a2f-444cb39b0c0e" />
or even:
<img width="834" height="460" alt="Untitled 8" src="https://github.com/user-attachments/assets/838cac3b-4de2-4e33-8486-1b5372b7a9c1" />